src/main/java/sentinel/storage/FileWriter.java

```diff
@@ -0,0 +1,82 @@
+package sentinel.storage;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+import sentinel.task.Deadline;
+import sentinel.task.Event;
+import sentinel.task.Task;
+import sentinel.utils.SentinelList;
+
+
+/**
+ * The FileWriter class is responsible for saving tasks from a SentinelList to a properties file.
+ * It writes the tasks to a file named "tasks.properties" in the "sentinel-saves" directory
+ * in a format that can be read by the FileLoader.
+ */
+public class FileWriter {
+    private static final Path DIRECTORY_PATH = Paths.get("sentinel-saves");
+    private static final Path FILE_PATH = DIRECTORY_PATH.resolve("tasks.properties");
+
+    private final SentinelList arrayList;
```

Comment:
arrayList is probably not the best name choice here 😅 . As above, `userTasks` or a more descriptive name could be helpful :)